app/assets/javascripts/app/ong-registration.js

```diff
@@ -0,0 +1,66 @@
+(function($) {
+
+
+  $('#new_ong').on('submit', function( event ) {
+    var nombre = validarNoVacio($('#ong_name'));
+    var direccion = validarNoVacio($('#ong_address'));
+    var codigoPostal = validarNoVacio($('#ong_zip'));
+    var telefono = validarNoVacio($('#ong_phone'));
+    var emailValido = validarEmail();
+    return nombre && direccion && codigoPostal && telefono && emailValido;
+  })
+
+  function validarEmail(){
+    var email = $('#ong_adminEmail');
+    if ( email.val() === '' ){
+      email.next().removeClass('texto-ayuda');
+      email.next().addClass('texto-validacion');
+      email.addClass('tiene-error');
+      email.next().html('Este campo es obligatorio');
+      return false;
+    }
+    if ((/^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i).test( email.val())){
+      email.removeClass('tiene-error');
+      email.next().html('&nbsp;');
+      return true;
+    }else{
+      email.addClass('tiene-error');
+      email.next().html('Email incorrecto');
+      return false;
+    }
+  }
+
+  function validarNoVacio(nombre){
+    if ( nombre.val() === '' ){
+      nombre.addClass('tiene-error');
+      if (nombre.next().hasClass('texto-validacion')){
+        nombre.next().html('Este campo es obligatorio');
+      }else{
+        nombre.nextAll('span:first').html('Este campo es obligatorio');
+      }
+      return false;
+    }
+    nombre.removeClass('tiene-error');
+    nombre.next().html('&nbsp;');
+    return true;
+  }
+
+  $('#ong_name').on('blur keyup', function() {
+    validarNoVacio($('#ong_name'));
+  });
+
+  $('#ong_address').on('blur keyup', function() {
+    validarNoVacio($('#ong_address'));
+  });
+
+  $('#ong_zip').on('blur keyup', function() {
+    validarNoVacio($('#ong_zip'));
+  });
+
+  $('#ong_phone').on('blur keyup', function() {
+    validarNoVacio($('#ong_phone'));
+  });
+
+  $('#ong_adminEmail').on('blur keyup', validarEmail);
+
+}(window.jQuery));
```

Comment:
Hola Fernando,

Ya hice las correcciones excepto: $('#ong_name').on('blur keyup',
validarNoVacio);
Porque la función validarNoVacio también la debo usar cuando se hace el
submit del formulario, y ahí no existen los eventos para utilizar en
la función.

Saludos,
Carolina

2015-10-22 19:43 GMT-03:00 Fernando Fajardo <notifications@github.com>:

> In app/assets/javascripts/app/ong-registration.js
> <https://github.com/OurMark/hermanosdesangre/pull/47#discussion_r42815762>
> :
>
> > +
> > +  $('#ong_address').on('blur keyup', function() {
> > +    validarNoVacio($('#ong_address'));
> > +  });
> > +
> > +  $('#ong_zip').on('blur keyup', function() {
> > +    validarNoVacio($('#ong_zip'));
> > +  });
> > +
> > +  $('#ong_phone').on('blur keyup', function() {
> > +    validarNoVacio($('#ong_phone'));
> > +  });
> > +
> > +  $('#ong_adminEmail').on('blur keyup', validarEmail);
> > +
> > +}(window.jQuery));
>
> cada archivo js debe tener al final un salto de linea.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OurMark/hermanosdesangre/pull/47/files#r42815762>.
>

-- 


The information contained in this e-mail may be confidential. It has been 
sent for the sole use of the intended recipient(s). If the reader of this 
message is not an intended recipient, you are hereby notified that any 
unauthorized review, use, disclosure, dissemination, distribution or 
copying of this communication, or any of its contents, 
is strictly prohibited. If you have received it by mistake please let us 
know by e-mail immediately and delete it from your system. Many thanks.

 

La información contenida en este mensaje puede ser confidencial. Ha sido 
enviada para el uso exclusivo del destinatario(s) previsto. Si el lector de 
este mensaje no fuera el destinatario previsto, por el presente queda Ud. 
notificado que cualquier lectura, uso, publicación, diseminación, 
distribución o copiado de esta comunicación o su contenido está 
estrictamente prohibido. En caso de que Ud. hubiera recibido este mensaje 
por error le agradeceremos notificarnos por e-mail inmediatamente y 
eliminarlo de su sistema. Muchas gracias.